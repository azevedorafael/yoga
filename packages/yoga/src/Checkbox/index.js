import Switch from './web/Switch';

const Checkbox = {};

Checkbox.Switch = Switch;

export default Checkbox;
