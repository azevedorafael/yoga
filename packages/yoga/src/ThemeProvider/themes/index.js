import corp from './corp';
import endUser from './end-user';
import gyms from './gyms';

export { endUser as default, endUser, corp, gyms };
