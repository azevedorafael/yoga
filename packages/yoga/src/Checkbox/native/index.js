import Switch from './Switch';
import Checkbox from './Checkbox';

export { Checkbox, Switch };
