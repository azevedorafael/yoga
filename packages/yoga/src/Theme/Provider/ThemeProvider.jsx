import React from 'react';
import { ThemeProvider as SCThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import * as tokens from '@gympass/yoga-tokens';
import * as themes from '../themes';

const getTheme = ({ theme, locale }) => {
  const token = tokens[locale] || tokens.default;
  const appTheme = themes[theme] || themes.default;

  return appTheme(token);
};

const ThemeProvider = ({ children, ...theme }) => (
  <SCThemeProvider theme={{ yoga: getTheme(theme) }}>
    {children}
  </SCThemeProvider>
);

ThemeProvider.propTypes = {
  theme: PropTypes.string,
  locale: PropTypes.string,
  children: PropTypes.node.isRequired,
};

ThemeProvider.defaultProps = {
  theme: 'endUser',
  locale: 'pt-BR',
};

export default ThemeProvider;
