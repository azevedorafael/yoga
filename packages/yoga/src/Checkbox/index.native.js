import { Checkbox, Switch } from './native';

Checkbox.Switch = Switch;

export default Checkbox;
