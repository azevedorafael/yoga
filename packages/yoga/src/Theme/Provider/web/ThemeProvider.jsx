import React from 'react';
import { node } from 'prop-types';

import Provider from '../Provider';
import GlobalStyle from './GlobalStyle';

/** This component provides a theme to all React components underneath itself via the context API. */
const ThemeProvider = ({ children, ...rest }) => (
  <Provider {...rest}>
    <GlobalStyle />
    {children}
  </Provider>
);

ThemeProvider.propTypes = {
  children: node.isRequired,
};

export default ThemeProvider;
