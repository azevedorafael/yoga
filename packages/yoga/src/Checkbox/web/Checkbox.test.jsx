import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Checkbox from '..';
import ThemeProvider from '../../ThemeProvider';

const data = {
  value: 'value',
  label: 'Checkbox Component',
  helper: 'Helper Text',
};

/* eslint-disable no-console */
const originalError = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = originalError;
});

describe('<Checkbox />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with default checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with disabled checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} disabled />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with checked checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} checked />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with disabled and checked checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} disabled checked />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with error checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} error />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with error and checked checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} error checked />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChange function when press on Checkbox', () => {
      const onChangeMock = jest.fn();
      const { getByRole } = render(
        <ThemeProvider>
          <Checkbox {...data} onChange={onChangeMock} />
        </ThemeProvider>,
      );
      fireEvent.click(getByRole('checkbox', { hidden: true }));
      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
