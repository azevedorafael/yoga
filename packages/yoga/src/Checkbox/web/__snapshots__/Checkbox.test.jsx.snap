// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Snapshots should match snapshot with all variant styles 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #F46152;
}

.c9 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #652CDB;
}

.c11 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #2B9486;
}

.c12 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #F46152;
  background-color: #F46152;
}

.c12 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  fill: #FFFFFF;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c13 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #652CDB;
  background-color: #652CDB;
}

.c13 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  fill: #FFFFFF;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c14 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #2B9486;
  background-color: #2B9486;
}

.c14 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  fill: #FFFFFF;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

.c8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 .c2:active .c4,
.c8:focus-within .c4,
.c8:hover .c4 {
  display: block;
}

.c8 .c2:active .c4 {
  background-color: rgba(160,124,233,0.75);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.75);
}

.c8:focus-within .c4,
.c8:hover .c4 {
  background-color: rgba(160,124,233,0.5);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.5);
}

.c10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 .c2:active .c4,
.c10:focus-within .c4,
.c10:hover .c4 {
  display: block;
}

.c10 .c2:active .c4 {
  background-color: rgba(88,194,180,0.75);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.75);
}

.c10:focus-within .c4,
.c10:hover .c4 {
  background-color: rgba(88,194,180,0.5);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.5);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
        />
        <input
          class="c7"
          type="checkbox"
        />
      </label>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c8"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c9"
        />
        <input
          class="c7"
          type="checkbox"
        />
      </label>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c10"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c11"
        />
        <input
          class="c7"
          type="checkbox"
        />
      </label>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c12"
        >
          <svg />
        </div>
        <input
          checked=""
          class="c7"
          type="checkbox"
        />
      </label>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c8"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c13"
        >
          <svg />
        </div>
        <input
          checked=""
          class="c7"
          type="checkbox"
        />
      </label>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c10"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c14"
        >
          <svg />
        </div>
        <input
          checked=""
          class="c7"
          type="checkbox"
        />
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot with checked checkbox 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #F46152;
  background-color: #F46152;
}

.c6 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  fill: #FFFFFF;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

.c10 .c2:active .c4,
.c10:focus-within .c4,
.c10:hover .c4 {
  display: block;
}

.c10 .c2:active .c4 {
  background-color: rgba(160,124,233,0.75);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.75);
}

.c10:focus-within .c4,
.c10:hover .c4 {
  background-color: rgba(160,124,233,0.5);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.5);
}

.c11 .c2:active .c4,
.c11:focus-within .c4,
.c11:hover .c4 {
  display: block;
}

.c11 .c2:active .c4 {
  background-color: rgba(88,194,180,0.75);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.75);
}

.c11:focus-within .c4,
.c11:hover .c4 {
  background-color: rgba(88,194,180,0.5);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.5);
}

.c8 {
  width: 100%;
  margin-top: 4px;
}

.c9 {
  color: #9898A6;
  font-size: 14px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
        >
          <svg />
        </div>
        <input
          checked=""
          class="c7"
          type="checkbox"
        />
        Checkbox Component
      </label>
    </div>
    <div
      class="c8"
    >
      <span
        class="c9"
      >
        Helper Text
      </span>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot with default checkbox 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #F46152;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

.c10 .c2:active .c4,
.c10:focus-within .c4,
.c10:hover .c4 {
  display: block;
}

.c10 .c2:active .c4 {
  background-color: rgba(160,124,233,0.75);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.75);
}

.c10:focus-within .c4,
.c10:hover .c4 {
  background-color: rgba(160,124,233,0.5);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.5);
}

.c11 .c2:active .c4,
.c11:focus-within .c4,
.c11:hover .c4 {
  display: block;
}

.c11 .c2:active .c4 {
  background-color: rgba(88,194,180,0.75);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.75);
}

.c11:focus-within .c4,
.c11:hover .c4 {
  background-color: rgba(88,194,180,0.5);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.5);
}

.c8 {
  width: 100%;
  margin-top: 4px;
}

.c9 {
  color: #9898A6;
  font-size: 14px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
        />
        <input
          class="c7"
          type="checkbox"
        />
        Checkbox Component
      </label>
    </div>
    <div
      class="c8"
    >
      <span
        class="c9"
      >
        Helper Text
      </span>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot with disabled and checked checkbox 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #9898A6;
  background-color: #F46152;
  background-color: #9898A6;
}

.c6 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  fill: #FFFFFF;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

.c10 .c2:active .c4,
.c10:focus-within .c4,
.c10:hover .c4 {
  display: block;
}

.c10 .c2:active .c4 {
  background-color: rgba(160,124,233,0.75);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.75);
}

.c10:focus-within .c4,
.c10:hover .c4 {
  background-color: rgba(160,124,233,0.5);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.5);
}

.c11 .c2:active .c4,
.c11:focus-within .c4,
.c11:hover .c4 {
  display: block;
}

.c11 .c2:active .c4 {
  background-color: rgba(88,194,180,0.75);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.75);
}

.c11:focus-within .c4,
.c11:hover .c4 {
  background-color: rgba(88,194,180,0.5);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.5);
}

.c8 {
  width: 100%;
  margin-top: 4px;
}

.c9 {
  color: #9898A6;
  font-size: 14px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
          disabled=""
        >
          <svg />
        </div>
        <input
          checked=""
          class="c7"
          disabled=""
          type="checkbox"
        />
        Checkbox Component
      </label>
    </div>
    <div
      class="c8"
    >
      <span
        class="c9"
      >
        Helper Text
      </span>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot with disabled checkbox 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #9898A6;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

.c10 .c2:active .c4,
.c10:focus-within .c4,
.c10:hover .c4 {
  display: block;
}

.c10 .c2:active .c4 {
  background-color: rgba(160,124,233,0.75);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.75);
}

.c10:focus-within .c4,
.c10:hover .c4 {
  background-color: rgba(160,124,233,0.5);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.5);
}

.c11 .c2:active .c4,
.c11:focus-within .c4,
.c11:hover .c4 {
  display: block;
}

.c11 .c2:active .c4 {
  background-color: rgba(88,194,180,0.75);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.75);
}

.c11:focus-within .c4,
.c11:hover .c4 {
  background-color: rgba(88,194,180,0.5);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.5);
}

.c8 {
  width: 100%;
  margin-top: 4px;
}

.c9 {
  color: #9898A6;
  font-size: 14px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
          disabled=""
        />
        <input
          class="c7"
          disabled=""
          type="checkbox"
        />
        Checkbox Component
      </label>
    </div>
    <div
      class="c8"
    >
      <span
        class="c9"
      >
        Helper Text
      </span>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot with error and checked checkbox 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #F46152;
  background-color: #F46152;
  border-color: #CB3530;
  background-color: #CB3530;
}

.c6 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  fill: #FFFFFF;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

.c10 .c2:active .c4,
.c10:focus-within .c4,
.c10:hover .c4 {
  display: block;
}

.c10 .c2:active .c4 {
  background-color: rgba(160,124,233,0.75);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.75);
}

.c10:focus-within .c4,
.c10:hover .c4 {
  background-color: rgba(160,124,233,0.5);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.5);
}

.c11 .c2:active .c4,
.c11:focus-within .c4,
.c11:hover .c4 {
  display: block;
}

.c11 .c2:active .c4 {
  background-color: rgba(88,194,180,0.75);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.75);
}

.c11:focus-within .c4,
.c11:hover .c4 {
  background-color: rgba(88,194,180,0.5);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.5);
}

.c8 {
  width: 100%;
  margin-top: 4px;
}

.c9 {
  color: #CB3530;
  font-size: 14px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
        >
          <svg />
        </div>
        <input
          checked=""
          class="c7"
          type="checkbox"
        />
        Checkbox Component
      </label>
    </div>
    <div
      class="c8"
    >
      <span
        class="c9"
      >
        Error msg
      </span>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot with error checkbox 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #F46152;
  border-color: #CB3530;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

.c10 .c2:active .c4,
.c10:focus-within .c4,
.c10:hover .c4 {
  display: block;
}

.c10 .c2:active .c4 {
  background-color: rgba(160,124,233,0.75);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.75);
}

.c10:focus-within .c4,
.c10:hover .c4 {
  background-color: rgba(160,124,233,0.5);
  box-shadow: 0 0 0 8px rgba(160,124,233,0.5);
}

.c11 .c2:active .c4,
.c11:focus-within .c4,
.c11:hover .c4 {
  display: block;
}

.c11 .c2:active .c4 {
  background-color: rgba(88,194,180,0.75);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.75);
}

.c11:focus-within .c4,
.c11:hover .c4 {
  background-color: rgba(88,194,180,0.5);
  box-shadow: 0 0 0 8px rgba(88,194,180,0.5);
}

.c8 {
  width: 100%;
  margin-top: 4px;
}

.c9 {
  color: #CB3530;
  font-size: 14px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
        />
        <input
          class="c7"
          type="checkbox"
        />
        Checkbox Component
      </label>
    </div>
    <div
      class="c8"
    >
      <span
        class="c9"
      >
        Error msg
      </span>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot without props 1`] = `
.c7 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
}

.c0 * {
  box-sizing: border-box;
}

.c6 {
  position: relative;
  border-style: solid;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  border-radius: 4px;
  border-width: 2px;
  border-color: #F46152;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 14px;
  color: #41414A;
}

.c5 {
  position: absolute;
  display: none;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2:active .c4,
.c1:focus-within .c4,
.c1:hover .c4 {
  display: block;
}

.c1 .c2:active .c4 {
  background-color: rgba(250,185,178,0.75);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.75);
}

.c1:focus-within .c4,
.c1:hover .c4 {
  background-color: rgba(250,185,178,0.5);
  box-shadow: 0 0 0 8px rgba(250,185,178,0.5);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span
          class="c4 c5"
        />
        <div
          class="c6"
        />
        <input
          class="c7"
          type="checkbox"
        />
      </label>
    </div>
  </div>
</div>
`;
